lib/train/extras/os_detect_linux.rb

```diff
@@ -16,91 +16,117 @@ module DetectLinux # rubocop:disable Metrics/ModuleLength
       elsif !(raw = get_config('/etc/system-release')).nil?
         # Amazon Linux
-        @platform[:family] = redhatish_platform(raw)
+        @platform[:name] = redhatish_platform(raw)
         @platform[:release] = redhatish_version(raw)
       elsif !(suse = get_config('/etc/SuSE-release')).nil?
         version = suse.scan(/VERSION = (\d+)\nPATCHLEVEL = (\d+)/).flatten.join('.')
         version = suse[/VERSION = ([\d\.]{2,})/, 1] if version == ''
         @platform[:release] = version
-        @platform[:family] = 'suse'
-        @platform[:family] = 'opensuse' if suse =~ /^openSUSE/
+        @platform[:name] =  if suse =~ /^openSUSE/
+                              'opensuse'
+                            else
+                              'suse'
+                            end
       elsif !(raw = get_config('/etc/arch-release')).nil?
-        @platform[:family] = 'arch'
+        @platform[:name] = 'arch'
         # Because this is a rolling release distribution,
         # use the kernel release, ex. 4.1.6-1-ARCH
         @platform[:release] = uname_r
       elsif !(raw = get_config('/etc/slackware-version')).nil?
-        @platform[:family] = 'slackware'
+        @platform[:name] = 'slackware'
         @platform[:release] = raw.scan(/(\d+|\.+)/).join
       elsif !(raw = get_config('/etc/exherbo-release')).nil?
-        @platform[:family] = 'exherbo'
+        @platform[:name] = 'exherbo'
         # Because this is a rolling release distribution,
         # use the kernel release, ex. 4.1.6
         @platform[:release] = uname_r
       elsif !(raw = get_config('/etc/gentoo-release')).nil?
-        @platform[:family] = 'gentoo'
+        @platform[:name] = 'gentoo'
         @platform[:release] = raw.scan(/(\d+|\.+)/).join
       elsif !(raw = get_config('/etc/alpine-release')).nil?
-        @platform[:family] = 'alpine'
+        @platform[:name] = 'alpine'
         @platform[:release] = raw.strip
       elsif !(raw = get_config('/etc/coreos/update.conf')).nil?
-        @platform[:family] = 'coreos'
+        @platform[:name] = 'coreos'
         meta = lsb_config(raw)
         @platform[:release] = meta[:release]
       elsif !(os_info = fetch_os_release).nil?
         if os_info['ID_LIKE'] =~ /wrlinux/
-          @platform[:family]  = 'wrlinux'
+          @platform[:name] = 'wrlinux'
           @platform[:release] = os_info['VERSION']
         end
       end
 
+      @platform[:family] = family_for_platform
+
       !@platform[:family].nil? && !@platform[:release].nil?
     end
 
+    DEBIAN_FAMILY = %w{debian ubuntu linuxmint raspbian}.freeze
+    RHEL_FAMILY = %w{centos redhat oracle scientific enterpriseenterprise amazon xenserver cloudlinux ibm_powerkvm nexus_centos wrlinux}.freeze
+    SUSE_FAMILY = %w{suse opensuse}.freeze
+
+    def family_for_platform
+      if DEBIAN_FAMILY.include?(@platform[:name])
+        'debian'
+      elsif RHEL_FAMILY.include?(@platform[:name])
+        'rhel'
```

Comment:
We need to be compatible with serverspec here: http://serverspec.org/advanced_tips.html They use redhat.